protected/src/serde.rs

```diff
@@ -0,0 +1,39 @@
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
```

Comment:
There is no need to implement Serialize and Deserialize manually, you can add a conditional attribute to the struct definition which will serialize the struct by converting it to another type and serializing that:

```rust
#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
#[cfg_attr(feature = "serde", serde(from = "Vec<u8>", into = "Vec<u8>"))]
pub struct Protected(Memzero<Vec<u8>>); 
```

And that will give you the wanted behaviour, as long as `Protected` implements `From<Vec<u8>>`, `Clone` and `Into<Vec<u8>>`.

Reply:
I think I won't address this, because it requires [`Into<Vec<u8>> for Protected`](https://serde.rs/container-attrs.html#into) and that's something I'm trying to avoid since I want to keep ownership of those bytes to later zero-them. I mean, I actually removed the `ProtectedString::to_string`, so if the user really wants to *unprotect* those bytes it has to call `protected.as_ref().to_vec()` or `protected_str.as_ref().to_string()`.